lib/pbench/cli/server/shell.py

```diff
@@ -37,6 +40,57 @@ def find_the_unicorn(logger: Logger):
             )
 
 
+def keycloak_connection(config: PbenchServerConfig, logger: Logger):
+    """
+     Checks if the Keycloak server is up and accepting the connections.
+     The connection check does the GET request on the oidc server /health
+     endpoint and the sample response returned by the /health endpoint looks
+     like the following:
+        {
+            "status": "UP", # if the server is up
+            "checks": []
+        }
+     Note: The Keycloak server need to be started with health-enabled on.
+     Args:
+        config: PbenchServerConfig
+        logger: logger
+    Returns:
+        0 if successful
+    """
+    try:
+        oidc_server = str(config.get("authentication", "internal_server_url"))
+    except (NoSectionError):
+        logger.exception("Error fetching required configuration")
+        sys.exit(1)
+    except (NoOptionError):
+        try:
+            oidc_server = str(config.get("authentication", "server_url"))
+        except (NoOptionError):
+            logger.exception("Error fetching required configuration")
+            sys.exit(1)
+
```

Comment:
Alternately,
```
    oidc_server = config.get(
        "authentication",
        "internal_server_url",
        fallback=config.get(
            "authentication",
            "server_url",
            fallback="",
        ),
    )
    if not oidc_server:
        logger.error("Error fetching Keycloak configuration")
        sys.exit(1)
```
Or, if you really love the `try` block:
```
    try:
        oidc_server = config.get(
            "authentication",
            "internal_server_url",
            fallback=config.get("authentication","server_url"),
        )
    except (NoSectionError):
        logger.exception("Bad config file:  missing 'authentication' section")
        sys.exit(1)
    except (NoOptionError):
        logger.error("Bad config file:  no 'server_url' in  'authentication' section")
        sys.exit(1)
```
You shouldn't need the `str()` calls, since the values aren't numbers, right?